CONTRIBUTING.md

```diff
@@ -0,0 +1,292 @@
+
+
+## Contents
+
+[How to Contribute](#how-to-contribute)
+
+[Contributing an Alternative Implementation](#contributing-an-alternative-implementation)
+
+[Branch Management](#branch-management)
+
+[Pull Request Rules and Guidelines](#pull-request-rules-and-guidelines)
+
+[Fortran Style Guide](#fortran-style-guide)
+
+## How to Contribute
+
+Contributions to c_sw will be accepted via [pull request](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request),
+either from branches in this repository, or from a fork of this repository. Pull
+requests will be reviewed and evaluated based on the technical merit of the
+proposed changes as well as conformance to the style guidelines outlined in this
+document. Code reviewers may request changes as a condition of acceptance of the
+pull request.
+
+## Contributing an Alternative Implementation
+
+Contributors who wish to provide an alternative implementation of `c_sw`
+must do so by issuing a pull request to the `develop` branch from their
+own forks of this repository. Write access to this repository can not be
+granted for external collaborators.
+
+All pull requests for new implementations must include:
+
+* A build system that functions across platforms and (if applicable) across
+multiple compilers.
+* Source code that is well written, properly formatted, and documented.
+* A test suite that passes and duplicates the tests that are run in the
+reference implementation. At present, 11 digits of accuracy are expected.
+* A GitHub Actions configuration to automate the new implementation's
+test suite.
+* A documentation README.md describing the new implementation, including how
+to run and test it.
+
+## Branch Management
+
+This repository follows the [GitHub Flow](https://guides.github.com/introduction/flow/)
+branching model with the following modifications borrowed from
+[Git Flow](https://nvie.com/posts/a-successful-git-branching-model/):
+
+* All branches that add new features, capabilities, or enhancements must be named:
+`feature/name-of-my-feature`
+* All branches that fix defects must be named: `bugfix/name-of-my-bugfix`
+
+In addition to the naming conventions, all branches must have a clearly defined,
+singular purpose, described by their name. It is also prefered, but not required,
+that branches correspond to an issue documented in the issue tracking system. (Issues
+can be added after branch creation, if necessary.)  All branches shall result in a
+pull request and shall be deleted immediately after that pull request is merged.
+
+## Pull Request Rules and Guidelines
+
+We ask contributors to please be mindful of the following when submitting a Pull
+Request. The following will help reviewers give meaningful and timely feedback.
+
+* Pull requests will not be accepted for branches that are not up-to-date with
+the current `develop` branch.
+* Pull requests will not be accepted unless all tests pass.
```

Comment:
Should we specify which tests here? Continuous Integration tests or kernel tests via running `ctest`?

Reply:
@robgonzalezpita - The `ctest` and CI tests are one and the same.  But we can clarify that.

Reply:
Fixed.